### PR TITLE
support animation of model overlays

### PR DESCRIPTION
**Pull request by ZappoMan**
This PR adds support for animation of Model Overlays. This used to work, but somehow got removed/broken in some refactor at some point. This uses the same properties and animation logic as entity animations.

Test Plan:
1) Basic smoke test. Compare against latest dev-download. Make sure in particular that model based overlays still function as normal. Things to try include: stylus on tablet, hand controller models (like when you pause in the HMD)
2) a) Run this client, in any domain (or even in the void)...
    b) go to /0.465893,-0.424502,0.0257262/0,0.697909,0,0.716186
    c) Using the JS Console... run this test script:
```
 var overlayID = Overlays.addOverlay("model",
        {
            name: "cards overlay",
            url: "https://hifi-content.s3.amazonaws.com/ozan/dev/anim/deck_of_cards/animated_cards_with_faces.fbx",
            position: { x: 0, y: 0, z: 0},
            animationSettings: {
                url: "https://hifi-content.s3.amazonaws.com/ozan/dev/anim/deck_of_cards/animated_cards_with_faces.fbx",
                fps: 30,
                running: true,
                loop: true
            }
        });
```
   
 d) you should see some cards animating... these are overlays.. they will not be visible to anyone else...

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3743/HighFidelity-Beta-PR11481-be7729b27694102de36de256e88d57ee663b5b9b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3743/HighFidelity-Beta-PR11481-be7729b27694102de36de256e88d57ee663b5b9b.exe

**Comment by YuppyPlays** [comment]
Testing complete:

Basic smoke test. Compare against latest dev-download. Make sure in particular that model based overlays still function as normal. Things to try include: stylus on tablet, hand controller models (like when you pause in the HMD) **Check, all seemed in order to me UI wise and stylus/tablet/hands, paused, etc)
a) Run this client, in any domain (or even in the void)... **Check**
b) go to /0.465893,-0.424502,0.0257262/0,0.697909,0,0.716186
c) Using the JS Console... run this test script: <Script here> **Check**
d) you should see some cards animating... these are overlays.. they will not be visible to anyone else... **Check**

**Pass**

File: interface/src/ui/overlays/ModelOverlay.cpp [hunk 1]
```diff
@@ -301,3 +376,138 @@ QString ModelOverlay::getName() const {
...
+        return;
+    }
+
+    auto now = usecTimestampNow();
+    auto interval = now - _lastAnimated;
+    _lastAnimated = now;
+    float deltaTime = (float)interval / (float)USECS_PER_SECOND;
+    _animationCurrentFrame += (deltaTime * _animationFPS);
+
+    {
```
Comment (sethalves) [inline interface/src/ui/overlays/ModelOverlay.cpp]: any reason for this code being in its own block?

File: interface/src/ui/overlays/ModelOverlay.cpp [hunk 2]
```diff
@@ -301,3 +376,138 @@ QString ModelOverlay::getName() const {
...
+        int index = _jointMapping[j];
+
+        if (index >= 0) {
+            glm::mat4 translationMat;
+
+            if (_animationAllowTranslation) {
+                if (index < translations.size()) {
+                    translationMat = glm::translate(translations[index]);
+                }
+            }
```
Comment (sethalves) [inline interface/src/ui/overlays/ModelOverlay.cpp]: spurious line-break

File: interface/src/ui/overlays/ModelOverlay.cpp [hunk 3]
```diff
@@ -301,3 +376,138 @@ QString ModelOverlay::getName() const {
...
+            else if (index < animationJointNames.size()) {
+                QString jointName = fbxJoints[index].name;
+
+                if (originalFbxIndices.contains(jointName)) {
+                    // Making sure the joint names exist in the original model the animation is trying to apply onto. If they do, then remap and get it's translation.
```
Comment (sethalves) [inline interface/src/ui/overlays/ModelOverlay.cpp]: here comes an s

File: interface/src/ui/overlays/ModelOverlay.cpp [hunk 4]
```diff
@@ -301,3 +376,138 @@ QString ModelOverlay::getName() const {
...
+                    int remappedIndex = originalFbxIndices[jointName] - 1; // JointIndeces seem to always start from 1 and the found index is always 1 higher than actual.
+                    translationMat = glm::translate(originalFbxJoints[remappedIndex].translation);
+                }
+            }
+            glm::mat4 rotationMat;
+            if (index < rotations.size()) {
+                rotationMat = glm::mat4_cast(fbxJoints[index].preRotation * rotations[index] * fbxJoints[index].postRotation);
+            }
```
Comment (sethalves) [inline interface/src/ui/overlays/ModelOverlay.cpp]: else on same line as }

Comment (ZappoMan) [reply interface/src/ui/overlays/ModelOverlay.cpp]: damn you VS!!!

File: interface/src/ui/overlays/ModelOverlay.cpp (hunk 3 above)
Comment (ZappoMan) [reply interface/src/ui/overlays/ModelOverlay.cpp]: hahah - copy from an old comment

File: interface/src/ui/overlays/ModelOverlay.cpp (hunk 1 above)
Comment (ZappoMan) [reply interface/src/ui/overlays/ModelOverlay.cpp]: copy and paste

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3758/HighFidelity-Beta-PR11481-9c81bc5479464e81889128d13dbdd8a20e7eff78.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3758/HighFidelity-Beta-PR11481-9c81bc5479464e81889128d13dbdd8a20e7eff78.exe

**Pull request merged**